src/testrel.jl

```diff
@@ -517,13 +532,33 @@ function _execute_test(
     engine::String,
     program::String,
     timeout_sec::Int64,
-    readonly::Bool,
+    readonly::Bool;
+    retry_number=1,
 )
     @debug("$name: Starting execution")
-    rsp = exec_async(context, schema, engine, program;
-            readtimeout = timeout_sec, readonly)
+    rsp = try
+        # Exec async really should return after 2-3 seconds
+        exec_async(context, schema, engine, program; readtimeout=30, readonly)
```

Comment:
`wait_until_done` has an equally long timeout for `get_transaction`. I'm not sure offhand but I would suspect that it should return as fast as exec_async (even if only to say the transaction isn't finished).

Reply:
it should probably return even faster than `exec_async` given it's just checking transaction metadata. I'll throw in a quick PR to decrease that.

Reply:
on second thought, that one clearly hasn't been an issue and would need more error-proofing to see the benefit of a reduced timeout, so not a quick PR incoming